nixos/modules/programs/regreet.nix

```diff
@@ -79,9 +79,11 @@ in
     };
 
     systemd.tmpfiles.settings."10-regreet" = let
+      user = config.services.greetd.settings.default_session.user;
+
       defaultConfig = {
-        user = "greeter";
-        group = config.users.users.${config.services.greetd.settings.default_session.user}.group;
+        inherit user;
+        group = config.users.users.${user}.group;
```

Comment:
While the user `greeter` is always declared with a group when the greetd service is enabled, the same cannot be said for an arbitrary user configured in `services.greetd.settings.default_session.user`. Because of the type of `systemd.tmpfiles.settings.<name>.<name>.<name>.group` not allowing `null`, I think this should either be put behind `mkIf (users.users.${user}.group != "")` or just dropped altogether, leaving the default `"-"`.

I don't know this service well though, if there are any serious issues with the user missing a group it should probably have an assertion instead.

```suggestion
```

Reply:
Thank you, I've added an assertion for the user and an if/else for the group.
Also rebased and formatted with nixfmt-rfc-style.